products/views.py

```diff
@@ -29,3 +36,111 @@ def home_page(request):
     }
 
     return render(request, "product/dashboard.html", context)
+
+
+@login_required
+def add_brand(request):
+    """ To add a new brand in brand model """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    context = {}
+    if request.method == "POST":
+        brand = AddBrandForm(request.POST, request.FILES)
+        if brand.is_valid():
+            brand.save()
+            messages.success(request, BrandFormSuccessMessages.NEW_BRAND_ADDED)
+        return redirect('view-brand')
+
+    brand_form = AddBrandForm()
+
+    context = {
+        "form": brand_form,
+        "heading": AdminPortalHeadings.ADD_BRAND
+    }
+    return render(request, "product/brand/add_brand.html", context)
+
+
+@login_required
+def update_brands(request, pk):
+    """ To update brand details """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    context = {}
+
+    selected_brand = get_object_or_404(Brand, id=pk)
+    if request.method == "POST":
+        brand_form = UpdateBrandForm(
+            request.POST, request.FILES, instance=selected_brand
+        )
+        if brand_form.is_valid():
+            brand_form.save()
+            messages.success(request, BrandFormSuccessMessages.BRAND_UPDATED)
+        return redirect('view-brand')
+
+    brand_form = UpdateBrandForm(instance=selected_brand)
+
+    context = {
+        "form": brand_form,
+        "heading": AdminPortalHeadings.UPDATE_BRAND
+    }
+
+    return render(request, "product/brand/update_brand.html", context)
+
+
+@login_required
+def view_brands(request):
+    """ To display all brands """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    brand = Brand.objects.all()
+    if request.GET.get('search'):
+        brand = brand.filter(name__icontains=request.GET.get('search'))
+
+    p = Paginator(brand, 3)
+    page_number = request.GET.get('page')
+    try:
+        page_obj = p.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = p.page(1)
+    except EmptyPage:
+        page_obj = p.page(p.num_pages)
+
+    context = {
+        'page_obj': page_obj,
+        'heading': AdminPortalHeadings.ALL_BRANDS
+    }
+    return render(request, 'product/brand/view_brands.html', context)
+
+
+@login_required
+def delete_brand(request, pk):
+    """ To delete a brand from model """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    brand = Brand.objects.get(id=pk)
+    product_has_brand = BrandProduct.objects.filter(brand=pk)
```

Comment:
@rinkishivani if you are using `product_has_brand`  only to check whether brand has products or not then you can get brand's products by brand object using reverse relation and check exists(). 

reference: https://docs.djangoproject.com/en/5.0/ref/models/relations/
cc: @lalitvasoya

Reply:
Done